### ISSUE TITLE
Classifiers not supporting variable length 

### ISSUE THREAD

**Issue by rshtirmer** [rtavenar/tslearn#94]
```
X = utils.to_time_series_dataset([[1, 2, 3, 4], [1, 2, 3], [2, 5, 6, 7, 8, 9]])
y = [1, 1, 0]
clf = TimeSeriesSVC(kernel="gak", gamma=.1, sz=X.shape[1], d=X.shape[2])
clf.fit(X, y)
OR
clf = neighbors.KNeighborsTimeSeriesClassifier(metric="dtw")
clf.fit(X, y)
```
Both methods result in the following error:
```
 File "main.py", line 77, in <module>
    clf.fit(X, y)
  File "/usr/local/lib/python3.7/site-packages/tslearn/neighbors.py", line 224, in fit
    return super(KNeighborsTimeSeriesClassifier, self).fit(X_, y)
  File "/usr/local/lib/python3.7/site-packages/sklearn/neighbors/base.py", line 891, in fit
    X, y = check_X_y(X, y, "csr", multi_output=True)
  File "/usr/local/lib/python3.7/site-packages/sklearn/utils/validation.py", line 756, in check_X_y
    estimator=estimator)
  File "/usr/local/lib/python3.7/site-packages/sklearn/utils/validation.py", line 573, in check_array
    allow_nan=force_all_finite == 'allow-nan')
  File "/usr/local/lib/python3.7/site-packages/sklearn/utils/validation.py", line 56, in _assert_all_finite
    raise ValueError(msg_err.format(type_err, X.dtype))
ValueError: Input contains NaN, infinity or a value too large for dtype('float64').
```

**Comment by mjip** [comment]
The SVC and KNN classifiers require the input data to be of equal length in order to fit properly. As a temporary fix, you could do some preprocessing on your own to replace the missing values: https://scikit-learn.org/stable/modules/impute.html#impute

**Comment by rtavenar** [comment]
This is now fixed in dev branch and forthcoming version 0.2
